orion/pkg/domains/class.go

```diff
@@ -0,0 +1,59 @@
+package domains
+
+import (
+	"database/sql"
+	"errors"
+	"regexp"
+	"time"
+)
+
+var TABLE_CLASSES = "classes"
+
+type Class struct {
+	Id         uint
+	CreatedAt  time.Time      `db:"created_at"`
+	UpdatedAt  time.Time      `db:"updated_at"`
+	DeletedAt  sql.NullTime   `db:"deleted_at"`
+	ProgramId  string         `db:"program_id" json:"programId"`
+	SemesterId string         `db:"semester_id" json:"semesterId"`
+	ClassKey   sql.NullString `db:"class_key" json:"classKey"`
```

Comment:
Ok cool. So @mzhang2021 actually bumped into this before. I wrote a long spiel about properties having type `sql.NullString`. It's weird because as you've experienced in the controller layer, you deal with `sql.NullString` even though we're in the controller.

So make `ClassKey` a string type here. In the Insert & Update methods in the repo layer, we will give the `db.Exec` statement a `sql.NullString` type. If there is a real value, let `sql.NullString` have `Valid: true` and the string itself. Otherwise, have `Valid` be false.

Here's an example:
```
room := sql.NullString{String: location.Room, Valid: location.Room != ""}
result, err := stmt.Exec(
		now,
		location.LocId,
		location.Street,
		location.City,
		location.State,
		location.Zipcode,
		room, <---- notice this is not just location.Room
		locId)
```

Refer to the slack channel for more background information.